src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java

```diff
@@ -692,10 +692,17 @@ else if (variableUsageExpressions.isEmpty()) {
                 if (isChild(currentStatementAst, variableAst)) {
                     variableUsageExpressions.add(currentStatementAst);
                 }
+                // Skip all Resource nodes which don't contain the variable
+                else if (currentStatementAst.getType() == TokenTypes.RESOURCE) {
+                    currentStatementAst = currentStatementAst.getNextSibling();
+                    continue;
+                }
                 // If expression doesn't contain variable and this variable
                 // hasn't been met yet, then distance + 1.
                 else if (variableUsageExpressions.isEmpty()
-                        && currentStatementAst.getType() != TokenTypes.VARIABLE_DEF) {
+                        && currentStatementAst.getType() != TokenTypes.VARIABLE_DEF
+                        && currentStatementAst.getType() != TokenTypes.TYPE
+                        && currentStatementAst.getType() != TokenTypes.MODIFIERS) {
```

Comment:
The regression report looks correct but I am skeptical about the logic present here. Why is it not modified at lines 605-654? Have you tried that approach?

Reply:
I combined several if conditions into the `isStatementVariableDefinition()` function.

Reply:
Combining is ok. 
But method name should describe what it does.

You do `!myfinctio(ast)) {distance++; }`

So your name should about ignoring of distance extension.
Name it as `isZeroDistanceToken`

Reply:
Thanks. The name makes more sense.